src/sentry/auth/superuser.py

```diff
@@ -275,12 +292,54 @@ def set_logged_in(self, user, current_datetime=None):
         request = self.request
         if current_datetime is None:
             current_datetime = timezone.now()
+
+        token = get_random_string(12)
+
+        if is_self_hosted() or not self._need_validation():
+            self._set_logged_in(
+                expires=current_datetime + MAX_AGE,
+                token=token,
+                user=user,
+                current_datetime=current_datetime,
+            )
+
+            logger.info(
+                "superuser.logged-in",
+                extra={"ip_address": request.META["REMOTE_ADDR"], "user_id": user.id},
+            )
+            return
+
+        try:
+            # need to use json loads as the data is no longer in request.data
+            su_access_json = json.loads(request.body)
+        except AttributeError:
```

Comment:
where does AtttributeError come from? how can this happen?

Reply:
i'll make this. more specific, this comes from the if there is an error from `json.loads(request.body)`

Reply:
you can put both the exceptions on the same line:
`except (json.JSONDecodeError, AttributeError)`.

Although if it is invalid json, wouldn't we want to return some sort of error to the client instead?